### PR TITLE
Maven publish plugin

### PR DESCRIPTION
**Pull request by kerkhofsd**
During development, we encountered some issues with the pom generated by the Gradle `maven` plugin. With this PR, we are moving to the 'standard' `maven-publish` plugin. 

FYI, from the Gradle documentation:
https://docs.gradle.org/current/userguide/maven_plugin.html

> This chapter describes deploying artifacts to Maven repositories using the original publishing mechanism available in Gradle 1.0: in Gradle 1.3 a new mechanism for publishing was introduced. This new mechanism introduces some new concepts and features that make Gradle publishing even more powerful and is now the preferred option for publishing artifacts.

### CODE REVIEW

File: alfresco-dynamic-extensions-repo/build.gradle [hunk 1]
```diff
@@ -4,13 +4,14 @@ subprojects {
     configurations {
         ampArtifact
         ampLib
-        compile.extendsFrom(ampLib)
+        compileOnly.extendsFrom(ampLib)
+        testCompile.extendsFrom(compileOnly)
```
Comment (vierbergenlars) [inline alfresco-dynamic-extensions-repo/build.gradle]: I'm not sure if you want to do it like this, adding everything from `compileOnly` to the `testCompile` configuration, as this adds more dependencies to the test runtime classpath than you may need.

Do you need your BOM on the `testCompile` configturation, or is it enough to have
```suggestion
        testCompileOnly.extendsFrom(compileOnly)
```

Comment (kerkhofsd) [reply alfresco-dynamic-extensions-repo/build.gradle]: I fully agree, but I followed the principles that are currently used in the rest of the project. 
(E.g. same `testCompile.extendsFrom(compileOnly)` in `annotations-runtime`.

I consider 'correcting' this out of scope...

What you suggest is ` testCompileOnly.extendsFrom(compileOnly)`
& explicitly importing the required `testRuntime` configurations?

Comment (vierbergenlars) [reply alfresco-dynamic-extensions-repo/build.gradle]: Hmm, does testRuntime require anything from the alfresco BOM?

Comment (kerkhofsd) [reply alfresco-dynamic-extensions-repo/build.gradle]: Apparently it does:
`* What went wrong:
Could not resolve all files for configuration ':alfresco-dynamic-extensions-repo:alfresco-dynamic-extensions-repo-50:testRuntimeClasspath'.
> Could not find org.alfresco:alfresco-repository:.
  Required by:
      project :alfresco-dynamic-extensions-repo:alfresco-dynamic-extensions-repo-50
> Could not find org.alfresco:alfresco-data-model:.
  Required by:
      project :alfresco-dynamic-extensions-repo:alfresco-dynamic-extensions-repo-50
`

**Pull request merged**